src/diffusers/pipelines/pag/pag_utils.py

```
@@ -25,123 +30,60 @@
-        Check if each layer input in `applied_pag_layers` is valid. It should be either one of these 3 formats:
-        "{block_type}", "{block_type}.{block_index}", or "{block_type}.{block_index}.{attention_index}". `block_type`
-        can be "down", "mid", "up". `block_index` should be in the format of "block_{i}". `attention_index` should be
-        in the format of "attentions_{j}".
-        """
-
-        layer_splits = layer.split(".")
-
-        if len(layer_splits) > 3:
-            raise ValueError(f"pag layer should only contains block_type, block_index and attention_index{layer}.")
-
-        if len(layer_splits) >= 1:
-            if layer_splits[0] not in ["down", "mid", "up"]:
-                raise ValueError(
-                    f"Invalid block_type in pag layer {layer}. Accept 'down', 'mid', 'up', got {layer_splits[0]}"
-                )
-
-        if len(layer_splits) >= 2:
-            if not layer_splits[1].startswith("block_"):
-                raise ValueError(f"Invalid block_index in pag layer: {layer}. Should start with 'block_'")
-
-        if len(layer_splits) == 3:
-            if not layer_splits[2].startswith("attentions_"):
-                raise ValueError(f"Invalid attention_index in pag layer: {layer}. Should start with 'attentions_'")
+    r"""Mixin class for [Pertubed Attention Guidance](https://arxiv.org/abs/2403.17377v1)."""
 
     def _set_pag_attn_processor(self, pag_applied_layers, do_classifier_free_guidance):
         r"""
         Set the attention processor for the PAG layers.
         """
-        if do_classifier_free_guidance:
-            pag_attn_proc = PAGCFGIdentitySelfAttnProcessor2_0()
+        pag_attn_processors = getattr(self, "_pag_attn_processors", None)
+        if pag_attn_processors is None:
+            # If this hasn't been set by the user, we default to the original PAG identity processors
+            pag_attn_processors = (PAGCFGIdentitySelfAttnProcessor2_0(), PAGIdentitySelfAttnProcessor2_0())
```

I think it's better to raise an error here